src/python/pants/backend/python/typecheck/mypy/rules_integration_test.py

```diff
@@ -204,21 +213,23 @@ def add(x: int, y: int) -> str:
             ),
         )
         self.add_to_build_file(
-            f"{self.package}/math", f"python_library(dependencies=['{self.package}/util'])",
```

Comment:
I don't think this is a good change iiuc. We are testing that the transitive dependency works properly on `util`. You only have dep inference on for `__init__.py`, so we no longer will declare the dep on `util:util`.

Reply:
Good call. Enabled inference.